build/Build.NuGet.Steps.cs

```diff
@@ -0,0 +1,163 @@
+        .Description(
+            "Builds the NuGet packages of the project assuming that any necessary build artifacts were already downloaded.")
+        .DependsOn(BuildManagedSrcNuGetPackages)
+        .DependsOn(CopyIntegrationsJsonForNuGetPackage)
+        .DependsOn(SetupRuntimeNativeFolderForNuGetPackage)
+        .DependsOn(BuildNuSpecNuGetPackages);
+
+    Target TestNuGetPackages => _ => _
+        .Description(
+            "Test the NuGet packages of the project assuming that the packages are available at bin/nuget-artifacts.")
+        .DependsOn(BuildNuGetPackagesTests)
+        .DependsOn(BuildNuGetPackagesTestApplications)
+        .DependsOn(RunNuGetPackagesTests);
+
+    Target BuildManagedSrcNuGetPackages => _ => _
+        .Description("Build the NuGet packages that are generated directly from src/**/*.csproj files")
+        .Executes(() =>
+        {
+            foreach (var project in Solution.GetManagedSrcProjects().Where(p => !p.Name.EndsWith("AdditionalDeps")))
+            {
+                DotNetPack(x => x
+                    .SetProject(project)
+                    .SetConfiguration(BuildConfiguration)
+                    .SetVersionSuffix(NuGetVersionSuffix)
+                    .SetOutputDirectory(NuGetArtifactsDirectory));
+            }
+        });
+
+    Target CopyIntegrationsJsonForNuGetPackage => _ => _
+        .Unlisted()
+        .Executes(() =>
+        {
+            var source = RootDirectory / "integrations.json";
+            var dest = RootDirectory / "nuget" / "OpenTelemetry.AutoInstrumentation" /
+                "contentFiles" / "any" / "any";
+            CopyFileToDirectory(source, dest, FileExistsPolicy.Overwrite);
+        });
+
+    Target SetupRuntimeNativeFolderForNuGetPackage => _ => _
+        .Unlisted()
+        .Description("Setup the \"runtimes/{platform}-{architecture}/native\" folders under \"nuget/OpenTelemetry.AutoInstrumentation.Runtime.Native\".")
+        .Executes(() =>
+        {
+            const string ciArtifactsDirectory = "bin/ci-artifacts";
+            const string baseRuntimeNativePath = "./nuget/OpenTelemetry.AutoInstrumentation.Runtime.Native/";
+
+            var requiredArtifacts = new string[]
+            {
+                "bin-alpine/linux-musl-x64",
+                "bin-centos/linux-x64",
+                "bin-macos-11/osx-x64",
+                "bin-windows-2022/win-x64",
+                "bin-windows-2022/win-x86"
+            };
+
+            foreach (var artifactFolder in requiredArtifacts)
+            {
+                var sourcePath = Path.Combine(ciArtifactsDirectory, artifactFolder);
+
+                var platformAndArchitecture = Path.GetFileName(artifactFolder);
+                var destinationPath =
+                    Path.Combine(baseRuntimeNativePath, "runtimes", platformAndArchitecture, "native");
+                DeleteDirectory(destinationPath);
+
+                CopyDirectoryRecursively(sourcePath, destinationPath);
```

Comment:
I got an exception when running `nuke BuildNuGetPackages` for the first time:

```
18:33:19 [ERR] Target SetupRuntimeNativeFolderForNuGetPackage has thrown an exception
System.ArgumentException: Expected directory to exist: bin/ci-artifacts\bin-alpine\linux-musl-x64 (Parameter 'source')
   at Nuke.Common.Assert.DirectoryExists(String path, String message, String argumentExpression) in /_/source/Nuke.Common/Assert.cs:line 172
   at Nuke.Common.IO.FileSystemTasks.CopyDirectoryRecursively(String source, String target, DirectoryExistsPolicy directoryPolicy, FileExistsPolicy filePolicy, Func`2 excludeDirectory, Func`2 excludeFile) in /_/source/Nuke.Common/IO/FileSystemTasks.cs:line 250
   at Build.<>c.<get_SetupRuntimeNativeFolderForNuGetPackage>b__40_1() in C:\c\opentelemetry-dotnet-instrumentation\build\Build.NuGet.Steps.cs:line 79
   at Nuke.Common.Execution.BuildExecutor.<>c.<Execute>b__4_2(Action x) in /_/source/Nuke.Common/Execution/BuildExecutor.cs:line 112
   at Nuke.Common.Utilities.Collections.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action) in /_/source/Nuke.Common/Utilities/Collections/Enumerable.ForEach.cs:line 17
   at Nuke.Common.Execution.BuildExecutor.Execute(NukeBuild build, ExecutableTarget target, IReadOnlyCollection`1 previouslyExecutedTargets, Boolean failureMode) in /_/source/Nuke.Common/Execution/BuildExecutor.cs:line 112
```

Am I missing something?

Reply:
Ops, I forgot to explain something that one needs to download the artifact to build the NuGet packages. We need the binaries from the different native builds. To build on your machine go to a successful CI run and download the bin-* zip files. Adding the automatic download from GH is probably a bit annoying, however, expanding them at the right location is something that we could add to our `nuke` targets. I added a small blurb to the `developing.md`, see cd6672e8c.